### DISCUSSION TITLE
Splitting Budgets into line items (feature suggestion)

### DISCUSSION THREAD

**Discussion by GabrielBarcia** [firefly-iii/firefly-iii#6230] [General]
First off I would like to say that I have been using for one year, and so far this is the software that has better suited me. Great job.

I would like to have a feature of splitting a budget into different line items. I do realize that this can be workaround with a tag/category/bill arrangement, but I think it would be beneficial to have it in a natural way

For a simple example of what I mean, can take an example of my current usage: Subscriptions

![image](https://user-images.githubusercontent.com/4810944/178682223-0a76812d-df4d-43b4-9d67-431aa78cb403.png)

I do have an individual budget per subscription, but It would be better if Subscriptions can be created as an individual budget, or as a container which total budget amount is equal to the sum of the lines. 

Same can happen with for example the Car, or the house. There are many different costs associated such as rent, loan payment, maintenances, insurance,  bills, but in the end is all tied to the big bucket House, Car, Subscriptions or whatever the container is for.

The benefit of this arrangement is the data can be seen, organized and analyzed in two different levels of details, when you can take either a bottom up approach, or a top down approach.

Hope you can consider it

Thanks and regards

**Comment by JC5** [comment]
What you want is already possible. What stops you from creating a budget called "Subscriptions" for (fictional amount) 500,- and adding multiple transactions to the budget?

There is no reason to create budgets for individual subscriptions. I don't see the advantage of that.

**Reply by GabrielBarcia** [reply]
Using Categories and Tags the data could be arranged in a way that resembles that, but the budget has this visualization were at a glance can be seen how are you doing in regards to budget, what was paid, what is missing, etc.

I can create a budget "Subscriptions" for an amount x. But if I add, or remove a subscription, then I need to manually go and change the fictional budget.

Regardless of the level of detail that the data wants to be tracked , there are two use cases with simple examples:

A bottom down summary, where the individual items can define the value of the budget container
![image](https://user-images.githubusercontent.com/4810944/178703640-e657f78d-7382-4522-9051-78a61711c69b.png)

Or a top down approach where the the total value od the container limits the amount of the sum of the lines
![image](https://user-images.githubusercontent.com/4810944/178704645-46554c97-b6ef-4114-89ae-4a80ea97ab25.png)

The advantage of using budgets rather than Categories or Tags is that the budget screen already available allow you two things: 1- have a clear view of the actual vs budget for whatever level of detail you want to track and 2- allow a simple, quick an visual analysis when you want to cut cost or reevaluate your money expenditure on where money is going and where to make changes. This same view is not available for Categories or tags.

**Reply by JC5** [reply]
OK. I'm happy that you like Firefly III, but I'm not really sure what your question is.

**Reply by GabrielBarcia** [reply]
My question, or to be more precise my ask, is if you can add a feature to the budgets.  

I can imagine two ways to do it
One way is to allow in the budget screen to create groups that I can assign the container (similar way as automated rules can be grouped) but that I can see a total budget for the group.

Other way is that when a budget is created you can chose if the budget be a single line (as it is today) or can be composed of multiple line items, each line item have his own budget and the container budget overall is equal to the sum of all the line items (as in the images I sent above)

Thanks :)

**Reply by JC5** [reply]
The option to group budgets is coming in a new release.

The latter is already possible, you can add many transactions to a single budget.

**Reply by GabrielBarcia** [reply]
Thank you James. I will wait then for the grouping of the budgets, that is good news.

Regarding to the other topic, when I say Line Item in the budget I don't mean transactions :) that is clear that can be done today. That I mean is a structure like

Budget
--> Sub-Budget Item 1
--> Sub-Budget Item 2
--> Sub-Budget Item 3

This structure organizes the assigned budget. During the course of a month, any number of transactions can be associated to each of the sub-budgets which will identify the actual expenditure. This is similar to grouping the budgets.

When you add the grouping feature, it would be very nice if can see a Subtotal value for the budget group :)

Keep up the good work. You have something very nice on the making.

**Reply by ttop207** [reply]
Hi James,
I'm a new user and love firefly already! I found this thread and had similar questions for things like Utilities. A parent utility budget with child budgets for water/sewer, electric, gas, etc. Did the grouping of budgets option come out in a release since this July 22 thread? I haven't seen it on the documentation or in the budget sections. 

Thanks for everything you've done and created! Its a fantastic system!

**Reply by GabrielBarcia** [reply]
What I end up doing was using the bills functionality to separate the individual bills, and have a ~ish budget. that has been working well for few years and have not changed back since then. IT separates the budget from the bill itself. Maybe it is helpful for you as well
